index.d.ts

```diff
@@ -59,8 +59,8 @@ export interface AxiosRequestConfig {
   responseType?: ResponseType;
   xsrfCookieName?: string;
   xsrfHeaderName?: string;
-  onUploadProgress?: (progressEvent: ProgressEvent) => void;
-  onDownloadProgress?: (progressEvent: ProgressEvent) => void;
+  onUploadProgress?: (progressEvent: { lengthComputable?: boolean; loaded?: number; total?: number }) => void;
```

Comment:
@jan-molak one small suggestion would be to declare a `ProgresEvent` interface (or similar if there are name clashes) with a comment explaining why is this required and maybe a link to the original spec or declaration.

Reply:
Thanks @sk- ! I have considered that, but eventually decided against it as calling such interface a ProgressEvent would cause a name clash with lib dom. I could call it AxiosProgressEvent if you like?

Reply:
I've updated my PR to address your feedback. Looking forward to hearing your thoughts!